src/alacris_lib/state_update.ml

```diff
@@ -6,75 +6,43 @@ open Types
 open Legilogic_ethereum
 open Operator_contract
 open Digesting
-open Ethereum_chain
+open Side_chain
 open Side_chain_operator
 open Side_chain_server_config
 
 let state_update_log = true
 
-let status_null_operation = ref 0
-
 let last_hash = ref Digest.zero
 let first_nontrivial_hash = ref Digest.zero
 
+let small_money_transfer : recipient:Address.t -> sender:Address.t -> unit Lwt_exn.t =
+  fun ~recipient ~sender ->
+  if state_update_log then
+    Logging.log "small_money_transfer_recipient_sender";
+  let open Lwt_exn in
+  let transfer_amount = TokenAmount.of_string "500000000000" in
+  let pre_transaction = Ethereum_user.transfer_tokens ~recipient transfer_amount in
+  Ethereum_user.post_pretransaction pre_transaction sender
+  >>= fun _ -> return ()
 
-let post_state_update : operator:Address.t -> operator_revision:Revision.t -> operator_digest:Digest.t -> unit Lwt_exn.t =
-  fun ~operator ~operator_revision ~operator_digest ->
+let post_state_update : operator:Address.t -> confirmed_state_update:StateUpdate.t -> unit Lwt_exn.t =
+  fun ~operator ~confirmed_state_update ->
   let open Lwt_exn in
   if state_update_log then
-    Logging.log "post_state_update operator_revision=%s digest=%s" (Revision.to_string operator_revision) (Digest.to_0x operator_digest);
-  if (String.equal (Digest.to_string !last_hash) (Digest.to_string operator_digest)) then
+    Logging.log "post_state_update confirmed_state_update.revision=%s digest=%s" (Revision.to_string confirmed_state_update.revision) (Digest.to_0x confirmed_state_update.state);
+  if (String.equal (Digest.to_string !last_hash) (Digest.to_string confirmed_state_update.state)) then
     (if state_update_log then
        Logging.log "Same hash as before. No need to do state_update. Instead doing small_money_transfer";
-     status_null_operation := 1 - !status_null_operation;
-     Side_chain_null_operation.small_money_transfer !status_null_operation
+     let heckle_address = Signing.Test.heckle_address in
+     small_money_transfer ~recipient:heckle_address ~sender:heckle_address
```

Comment:
If you're doing heckle to himself, no need for jeckle. Maybe Sysiphus, if you want to use a fixed prefunded account (versus the Croesus being the first account from geth).

Or you could restore the alternate Heckle vs Jeckle... I think it's funny, if slightly overkill.

Reply:
Jeckle has been eliminated.